enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/server/HttpRequestHandler.java

```diff
@@ -55,7 +62,11 @@ public ResponseObject process(CheckRequest request) {
         }
 
```

Comment:
need to set the ThreadContext.push after checking the null

Reply:
Fixed in https://github.com/wso2/product-microgateway/pull/2679/commits/356dd76f69e350e032432200814911fd84269ee1